omp/solver/gmres_kernels.cpp

```diff
@@ -76,9 +296,45 @@ void initialize_2(std::shared_ptr<const OmpExecutor> exec,
                   matrix::Dense<ValueType> *krylov_bases,
                   Array<size_type> *final_iter_nums, size_type krylov_dim)
 {
-    GKO_NOT_IMPLEMENTED;
+    for (size_type j = 0; j < residual->get_size()[1]; ++j) {
+        // Calculate residual norm
+        ValueType res_norm = zero<ValueType>();
+
+#pragma omp declare reduction(add : ValueType : omp_out = omp_out + omp_in)
+
+#pragma omp parallel for reduction(add : res_norm)
+        for (size_type i = 0; i < residual->get_size()[0]; ++i) {
+            res_norm += residual->at(i, j) * residual->at(i, j);
+        }
+        residual_norm->at(0, j) = sqrt(res_norm);
+
+#pragma omp parallel for
+        for (size_type i = 0; i < krylov_dim + 1; ++i) {
+            if (i == 0) {
+                residual_norm_collection->at(i, j) = residual_norm->at(0, j);
+            } else {
+                residual_norm_collection->at(i, j) = zero<ValueType>();
+            }
+        }
+
+#pragma omp parallel for
+        for (size_type i = 0; i < residual->get_size()[0]; ++i) {
+            krylov_bases->at(i, j) =
+                residual->at(i, j) / residual_norm->at(0, j);
+        }
+        final_iter_nums->get_data()[j] = 0;
+    }
+
+#pragma omp parallel for
+    for (size_type i = 0; i < krylov_bases->get_size()[0]; ++i) {
+        for (size_type j = residual->get_size()[1];
+             j < krylov_bases->get_size()[1]; ++j) {
+            krylov_bases->at(i, j) = zero<ValueType>();
+        }
+    }
 }
 
+
```

Comment:
This newline should be removed (to get the `INSTANTIATE` closer to the actual function).